ext/couchbase.cxx

```diff
@@ -6586,7 +6584,19 @@ cb_Backend_document_search(VALUE self, VALUE index_name, VALUE query, VALUE opti
         req.index_name = cb_string_new(index_name);
         req.query = cb_string_new(query);
 
-        cb_extract_option_bool(req.explain, options, "explain");
+        if (VALUE explain = rb_hash_aref(options, rb_id2sym(rb_intern("explain"))); !NIL_P(explain)) {
+            switch (TYPE(explain)) {
+                case T_TRUE:
+                    explain = true;
+                    break;
+                case T_FALSE:
+                    explain = false;
+                    break;
+                default:
+                    throw ruby_exception(rb_eArgError, rb_sprintf("explain must be a Boolean, but given %+" PRIsVALUE, explain));
+            }
+            req.explain = explain;
+        }
```

Comment:
why `cb_extract_option_bool` does not work here? it looks pretty much the same

https://github.com/couchbase/couchbase-ruby-client/blob/58fc25beb88260c23df0bf75e43a46288e28c547/ext/couchbase.cxx#L1802-L1821

Reply:
ah, because of `std::optional<bool>`?

Reply:
Then, lets make it template rather

```c++
template <typename Boolean>
static void
cb_extract_option_bool(Boolean& field, VALUE options, const char* name)
```